### PR TITLE
Test: add spin to test_requirements

### PR DESCRIPTION
**Pull request by renoauerbach**
I was following the documentation with context to running the tests for numpy: https://github.com/numpy/numpy/blob/fba4fc16a26e77ec4ce2b1900577f088b1d5a7a6/doc/source/dev/development_environment.rst?plain=1#L91-L95
Test requirements didn't indicate that spin was needed.  I amended test-requirements with spin.

### CODE REVIEW

**Comment by charris** [comment]
That makes a total of five places spin is installed, I wonder if we should have a `base_requirements.txt` file that contains it. Also, maybe an `all_requirements.txt` file.

File: doc/source/dev/development_environment.rst [hunk 1]
```diff
@@ -83,6 +83,7 @@ Before running the tests, first install the test dependencies::
 
     $ python -m pip install -r requirements/test_requirements.txt
     $ python -m pip install asv # only for running benchmarks
+    $ python -m pip install spin # utility for running the tests
```
Comment (mhvk) [inline doc/source/dev/development_environment.rst:86]: This would not be needed anymore with the change in `test_requirements.txt`, right?

**Review by mhvk — COMMENTED** [review]
Yes, thanks! Having to install `spin` caught me out several times.

**Review by rgommers — CHANGES_REQUESTED** [review]
`spin` is not a test requirement. Tests can be run on an installed `numpy` by (for example) `python -c "import numpy as np; np.test()"`.

What `spin` is is a _development_ requirement. The closest thing we have is `linter_requirements.txt`. We should probably have a `dev_requirements.txt` for this; not sure if `linter` should remain separate from `dev` or be included. Either way works for me.

I'm seeing too much wrong advice and confusion about `spin` recently, so this is kinda important. `spin` should only ever be used for working on numpy itself, not for anything else like regular installs or running tests.

File: doc/source/dev/development_environment.rst (hunk 1 above)
Comment (rgommers) [reply doc/source/dev/development_environment.rst:86]: Agreed, this change should be undone.

**Comment by rgommers** [comment]
Actually, `spin` is already included in `build_requirements.txt`, which in turn is included in `all_requirements.txt`. So this PR is unnecessary, there is nothing to do here. I will close it.

**Comment by rgommers** [comment]
> I was following the documentation with context to running the tests for numpy: 

For completeness, that document does include:
```
python -m pip install -r requirements/all_requirements.txt
```
so you simply missed a line when following the instructions I believe.

**Pull request closed without merging**